pkg/util/helper/service.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2025 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helper
+
+import (
+	"fmt"
+	"sort"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// DedupeAndSortServiceLoadBalancerIngress dedupes and sorts the ServiceLoadBalancerIngress slice.
+func DedupeAndSortServiceLoadBalancerIngress(ingresses []corev1.LoadBalancerIngress) []corev1.LoadBalancerIngress {
+	ipModeMap := make(map[corev1.LoadBalancerIPMode]*corev1.LoadBalancerIPMode)
+	errMap := make(map[string]*string)
+	for i, ingress := range ingresses {
+		if ingress.IPMode != nil {
+			// let same IPMode use same ptr, to let *IPMode be comparable
```

Comment:
Just a question: for IPMode, its enumeration value has been declared, so for the same value, its address is the same?

Reply:
I haven't tested this in a real environment, but the status is unmarshaled from the rb/crb status by the informer, so they should be at different addresses.
If the variables are declared in the code, they could be at the same address.

As a helper function, I think its implementation should not be affected by these scenarios.